aiohttp/connector.py

```diff
@@ -1121,6 +1126,9 @@ async def _wrap_create_connection(
             if exc.errno is None and isinstance(exc, asyncio.TimeoutError):
                 raise
             raise client_error(req.connection_key, exc) from exc
+        finally:
+            if sock is not None:
```

Comment:
Explicitly closing the sock is fine, we shouldn't have to do this but it's also fine.

Normally I'd prefer uvloop fix the race. However given how popular uvloop is, how high the impact can be, and the fact a fix is stalled upstream, I think it's reasonable to do the explicit close here.